admin_setting.go

```diff
@@ -0,0 +1,461 @@
+type adminSettings struct {
+	client *Client
+}
+
+// AdminGeneralSetting represents a the general settings in Terraform Enterprise.
+type AdminGeneralSetting struct {
+	ID                            string `jsonapi:"primary,general-settings"`
+	LimitUserOrganizationCreation bool   `jsonapi:"attr,limit-user-organization-creation"`
+	APIRateLimitingEnabled        bool   `jsonapi:"attr,api-rate-limiting-enabled"`
+	APIRateLimit                  int    `jsonapi:"attr,api-rate-limit"`
+	SendPassingStatusesEnabled    bool   `jsonapi:"attr,send-passing-statuses-for-untriggered-speculative-plans"`
+	AllowSpeculativePlansOnPR     bool   `jsonapi:"attr,allow-speculative-plans-on-pull-requests-from-forks"`
+}
+
+// GetGeneral returns the general settings.
+func (s *adminSettings) GetGeneral(ctx context.Context) (*AdminGeneralSetting, error) {
+	req, err := s.client.newRequest("GET", "admin/general-settings", nil)
+	if err != nil {
+		return nil, err
+	}
+
+	ags := &AdminGeneralSetting{}
+	err = s.client.do(ctx, req, ags)
+	if err != nil {
+		return nil, err
+	}
+
+	return ags, nil
+}
+
+// AdminGeneralSettingsUpdateOptions represents the admin options for updating
+// general settings.
+// https://www.terraform.io/docs/cloud/api/admin/settings.html#request-body
+type AdminGeneralSettingsUpdateOptions struct {
+	LimitUserOrgCreation              *bool `jsonapi:"attr,limit-user-organization-creation,omitempty"`
+	APIRateLimitingEnabled            *bool `jsonapi:"attr,api-rate-limiting-enabled,omitempty"`
+	APIRateLimit                      *int  `jsonapi:"attr,api-rate-limit,omitempty"`
+	SendPassingStatusUntriggeredPlans *bool `jsonapi:"attr,send-passing-statuses-for-untriggered-speculative-plans,omitempty"`
+	AllowSpeculativePlansOnPR         *bool `jsonapi:"attr,allow-speculative-plans-on-pull-requests-from-forks,omitempty"`
+}
+
+// UpdateGeneral updates the general settings.
+func (s *adminSettings) UpdateGeneral(ctx context.Context, options AdminGeneralSettingsUpdateOptions) (*AdminGeneralSetting, error) {
+	req, err := s.client.newRequest("PATCH", "admin/general-settings", &options)
+	if err != nil {
+		return nil, err
+	}
+
+	ags := &AdminGeneralSetting{}
+	err = s.client.do(ctx, req, ags)
+	if err != nil {
+		return nil, err
+	}
+
+	return ags, nil
+}
+
+// AdminCostEstimationSetting represents the admin cost estimation settings.
+type AdminCostEstimationSetting struct {
+	ID                  string `jsonapi:"primary,cost-estimation-settings"`
+	Enabled             bool   `jsonapi:"attr,enabled"`
+	AWSAccessKeyID      string `jsonapi:"attr,aws-access-key-id"`
+	AWSAccessKey        string `jsonapi:"attr,aws-secret-key"`
+	GCPCredentials      string `jsonapi:"attr,gcp-credentials"`
+	AzureClientID       string `jsonapi:"attr,azure-client-id"`
+	AzureClientSecret   string `jsonapi:"attr,azure-client-secret"`
+	AzureSubscriptionID string `jsonapi:"attr,azure-subscription-id"`
+	AzureTenantID       string `jsonapi:"attr,azure-tenant-id"`
```

Comment:
I think this (and Options) is missing `aws-instance-profile-enabled` ?

Reply:
Added